### DISCUSSION TITLE
How to use custom template for specific menuicons in p-menubar?

### DISCUSSION THREAD

**Discussion by ova2** [primefaces/community#998] [PrimeNG] [answered]
In my company, we have our own design system and own SVG icons. We generated SVG sprite and used it with Angulat Material. Now, we want to use SVG icons from sprite with PrimeNG. For instance with Menubar. It is possible to use custom template for **menuicon**. So, the idea is something like this:

```html
<p-menubar [model]="items">
    <ng-template pTemplate="menuicon">
        <svg>
          <use xlink:href="/path/to/icons.svg#some-icon-id"></use>
        </svg>
    </ng-template>
</p-menubar>
```

But there are many menu items in the model and it is not clear how to use individual icon per menu item. Is there a context accessible within the template, so that we can distinguish between individual menu item and take a specific SVG icon from sprite? There is no examples from PrimeNG and we can not figure out this from the documented API.

Thanks for any tip.

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

There are 3 options for that.

1-) `menuicon` template to customize the menu trigger icon (displayed in mobile)
2-) `submenuicon` template to customize the trigger icons of the menu items (default: arrow down icon)
3-) `item` template to customize the entire menu item. An example of the menu component is here, you can also implement the same code to the menubar.
